src/zcbor_decode.c

```diff
@@ -122,7 +122,7 @@ static void endian_copy(uint8_t *dst, const uint8_t *src, size_t src_len)
  *          big to little-endian if necessary (@ref ZCBOR_BIG_ENDIAN).
  */
 static bool value_extract(zcbor_state_t *state,
-		void *const result, size_t result_len)
+		void *const result, size_t result_len, bool *indefinite_length_array)
```

Comment:
Couldn't we take a struct pointer as parameter to both populate the value and communicate if the array length is indefinite? IMO this solution is prone to errors by callers passing NULL when they shouldn't.

Reply:
It's a static function, so there's a lot of trust. There are also asserts checking for NULL params (I added one for `state` now). I imagine using a struct will incur some extra code size.